net/vpnc/files/vpnc.sh

```diff
@@ -90,6 +91,11 @@ proto_vpnc_teardown() {
 
 	pwfile="/var/etc/vpnc-$config.conf"
 
+	json_get_var authfail authfail
+	if [ "${authfail:-0}" -gt 0 ]; then
+		proto_block_start "$config"
```

Comment:
On 16-03-12 10:16 PM, Yousong Zhou wrote:
> In net/vpnc/files/vpnc.sh
> <https://github.com/openwrt/packages/pull/2495#discussion_r55931568>:
>
>> @@ -90,6 +91,11 @@ proto_vpnc_teardown() {
>>
>>  	pwfile="/var/etc/vpnc-$config.conf"
>>
>> +	json_get_var authfail authfail
>> +	if [ "${authfail:-0}" -gt 0 ]; then
>> +		proto_block_start "$config"
>
> It's |proto_block_restart|. Apart from that, you need a way to know the

Oops, thanks.

> interface was tear down because of auth failure before doing this
> |authfail| config check. |pppd| signifies this condition in its exit
> status (i.e. |11| and |19|).

Hmmm... ok, I'll check if vpnc has a similar exit code thing.

>
> This feature is good and reasonable, thanks, but /personally/ I would
> expect it being requested/implemented if it really bothers someone to
> keep the feature set compact and slim. I myself came across such a
> situation using |pppd| against a local ISP :)
> (https://dev.openwrt.org/ticket/14144#comment:10)

I personally ran into the problem with vpnc authentication failures 
leading to an account lockout, hence my interest (but I don't use vpnc 
much, hence the lack of testing so far).

Regards,

Daniel